python/lsst/ts/ofc/scripts/generate_alpha_values.py

```diff
@@ -0,0 +1,271 @@
+
+
+def save_alpha_values(alpha_values: np.ndarray, path: str) -> None:
+    """Save alpha values to a yaml file.
+
+    Parameters
+    ----------
+    alpha_values : np.ndarray
+        Alpha values for the normalized point-source sensitivity.
+    path : str
+        Path to save the alpha values.
+    """
+
+    # Convert numpy array to a list,
+    # as yaml.dump cannot handle numpy arrays directly
+    alpha_values_list = alpha_values.tolist()
+
+    # Complete path to the yaml file
+    yaml_file_path = f"{path}/alpha_values.yaml"
+
+    # Write header and alpha values to the yaml file
+    with open(yaml_file_path, "w") as yaml_file:
+        header = (
+            "---\n"
+            "\n# Default alpha coefficient for the normalized point-source sensitivity\n"
+            "# Normalized point-source sensitivity (PSSN) information\n"
+            "# PSSN ~ 1 - alpha * delta^2\n"
+            "# Eq (7.1) in Normalized Point Source Sensitivity for LSST\n"
+            "# There are 25 terms of alpha value here for z4-z28 to use.\n"
+            "# The procedure to obtain them is detailed in scripts/generate_alpha_values.py\n"
+            "# For more details, please check the page 19-21 in:\n"
+            "# https://docushare.lsst.org/docushare/dsweb/Get/Document-18041/150513.pptx.pdf\n\n"
+        )
+        yaml_file.write(header)
+        yaml.dump(alpha_values_list, yaml_file)
+
+
+def get_alpha_values(
+    config: dict,
+    instrument: str,
+    band: str,
+    znmin: int = 4,
+    znmax: int = 28,
+) -> np.ndarray:
+    """Calculate alpha values for the normalized point-source sensitivity.
+
+    Parameters
+    ----------
+    config : dict
+        Configuration dictionary.
+    instrument : str
+        Instrument name.
+    band : str
+        Filter band.
+    znmin : int
+        Minimum Zernike index.
+    znmax : int
+        Maximum Zernike index.
+
+    Returns
+    -------
+    np.ndarray
+        Alpha values for the normalized point-source sensitivity.
+    """
+
+    # Set wavlength and number of zernikes used
+    num_zk = znmax - znmin + 1
+    wavelength = config["wavelengths"][band]
+
+    # Set the increments of zernikes used to do the fit
+    increments = np.array([2, 4, 6, 8, 10]) * wavelength * 1e-2
+
+    # Create OFCData, OPDMetrology and fiducial telescope
+    ofc_data = OFCData("lsst")
+    opd_metrology = OpdMetrology()
+
+    if band == "ref":
+        fiducial = batoid.Optic.fromYaml(f"{instrument.upper()}_g_500.yaml")
+    else:
+        fiducial = batoid.Optic.fromYaml(f"{instrument.upper()}_{band}.yaml")
+
+    # Get the field angles of the gaussian quadrature points
+    field_angles = [
+        ofc_data.gq_points[sensor] for sensor in range(len(ofc_data.gq_weights))
+    ]
+    field_x, field_y = zip(*field_angles)
+
+    # Load the gaussian quadrature weights
+    imqw = [ofc_data.gq_weights[sensor] for sensor in range(len(ofc_data.gq_weights))]
+    n_imqw = imqw / np.sum(imqw)
```

Comment:
This line assume the weights (`imqw`) are all positive values: 

```
 n_imqw = imqw / np.sum(imqw)
```

It is not very uncommon to have the negative weight in physics:
https://www.physicsforums.com/threads/negative-weights-in-weighted-averages-explained.429827/

You could consider to throw out an error if there is any negative weight or have a more robust way to deal with it in physics based on your context.